### ISSUE TITLE
Timeout async guard inappropriately triggers with F#

### ISSUE THREAD

**Issue by bradwilson** [xunit/xunit#2892]
Reported here: https://github.com/xunit/xunit/issues/2222#issuecomment-1956489613

Sample code that fails to run with 2.7.0:

```fsharp
module Tests

open System.Threading.Tasks
open Xunit

[<Fact(Timeout = 3000)>]
let TestMethod() =
    task {
        Task.Delay(10) |> Async.AwaitTask |> ignore
        Assert.Fail("Make sure things waited")
    }
```

Adding `:> Task` cast to the end of the task lambda is a temporary workaround:

```fsharp
module Tests

open System.Threading.Tasks
open Xunit

[<Fact(Timeout = 3000)>]
let TestMethod() =
    task {
        Task.Delay(10) |> Async.AwaitTask |> ignore
        Assert.Fail("Make sure things waited")
    } :> Task
```

**Comment by bradwilson** [comment]
The "culprit" here is the new `task` keyword, which I wasn't familiar with. Instead of returning an [`FSharpAsync<T>`](https://fsharp.github.io/fsharp-core-docs/reference/fsharp-control-fsharpasync-1.html), it returns [`Task<T>`](https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.task-1), presumably for better cross-language interop.

**Comment by bradwilson** [comment]
Available in v2 `2.7.1-pre.7`
Available in v3 `0.1.1-pre.378`

https://xunit.net/docs/using-ci-builds
